src/include/migraphx/match/gelu_tanh.hpp

```diff
@@ -36,31 +36,43 @@ template <class F>
 struct gelu_tanh_matcher
 {
     F f;
-    auto pow_fn() const { return f("pow")(used_once(), arg(1)(has_value(3.0f))); }
 
+    /// x ^ 3
+    auto pow_fn() const { return f("pow")(used_once(), arg(1)(has_value(3.0))); }
+
+    /// tanh( sqrt(2/M_PI) * (x + 0.044715 * x ^ 3 )
     auto tanh_fn() const
     {
-        return f("tanh")(
-            used_once(),
-            arg(0)(f("mul")(either_arg(0, 1)(has_value(sqrt(M_2_PI), 1e-3),
-                                             f("add")(any_arg(0, 1)(f("mul")(either_arg(0, 1)(
-                                                 has_value(0.044715f), pow_fn()))))))));
+        auto mul_const_pow    = f("mul")(either_arg(0, 1)(has_value(0.044715), pow_fn()));
+        auto add_any_mul   = f("add")(any_arg(0, 1)(mul_const_pow));
+        auto mul_SQRT2RPI_add = f("mul")(either_arg(0, 1)(has_value(sqrt(M_2_PI)), add_any_mul));
+        return f("tanh")(used_once(), arg(0)(mul_SQRT2RPI_add));
+    }
+
+    /// x * (0.5? + 0.5 * tanh( sqrt(2/M_PI) * (x? + 0.044715 * x? ^ 3) ) )
+    /// <item>? question mark means it doesn't explicitly match that item (anything will work)
+    auto matcher_v0() const
+    {
+        auto mul_half_tanh = f("mul")(either_arg(0, 1)(has_value(0.5), tanh_fn()));
+        auto add_any_mul   = f("add")(any_arg(0, 1)(mul_half_tanh));
+        return f("mul")(either_arg(0, 1)(any().bind("x"), add_any_mul));
     }
 
-    auto matcher() const
+    /// x * 0.5 * (1.0 + tanh( sqrt(2/M_PI) * (x + 0.044715 * x ^ 3) ) )
+    auto matcher_v1() const
     {
-        return f("mul")(used_once(),
-                        either_arg(0, 1)(any().bind("x"),
-                                         f("add")(any_arg(0, 1)(f("mul")(
-                                             either_arg(0, 1)(has_value(0.5f), tanh_fn()))))));
+        auto add_one_tanh = f("add")(used_once(), either_arg(0, 1)(has_value(1.0), tanh_fn()));
+        auto mul_half_x = f("mul")(used_once(), either_arg(0, 1)(has_value(0.5), any().bind("x")));
+        return f("mul")(either_arg(0, 1)(mul_half_x, add_one_tanh));
     }
 };
 } // namespace detail
 
 template <class F>
 auto gelu_tanh(F f)
 {
-    return detail::gelu_tanh_matcher<F>{f}.matcher();
+    return any_of(detail::gelu_tanh_matcher<F>{f}.matcher_v0(),
+                  detail::gelu_tanh_matcher<F>{f}.matcher_v1());
```

Comment:
You should just construct `gelu_tanh_matcher` and call a `matcher` method. Inside the `matcher` method you can do `any_of`.